app/src/main/java/app/passwordstore/ui/crypto/PasswordDialog.kt

```diff
@@ -25,11 +25,23 @@ class PasswordDialog : DialogFragment() {
 
   private val binding by unsafeLazy { DialogPasswordEntryBinding.inflate(layoutInflater) }
   private var isError: Boolean = false
+  private var clearCacheChecked: Boolean = true
 
   override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
     val builder = MaterialAlertDialogBuilder(requireContext())
     builder.setView(binding.root)
     builder.setTitle(R.string.password)
+
+    if (requireArguments().getBoolean(ENABLED_CACHE_ARG_EXTRA, false)) {
+      clearCacheChecked = requireArguments().getBoolean(CLEAR_CACHE_ARG_EXTRA)
+      builder.setMultiChoiceItems(
+        arrayOf(getString(R.string.clear_cached_password_on_screen_off)),
+        BooleanArray(1) { clearCacheChecked },
+      ) { _, _, isChecked ->
+        if (isChecked) clearCacheChecked = true else clearCacheChecked = false
+      }
```

Comment:
using `setMultiChoiceItems` here ends up being incorrect for multiple reasons: 1) This is just one option not multiple, 2) It puts the item above the password input field which looks pretty weird.

I'm reworking the implementation to put the checkbox directly in the layout rather than pipe it through `setMultiChoiceItems`.